From e30b13c08056630dbea6b4bbeee560e3df11a7dd Mon Sep 17 00:00:00 2001
From: dfffffff <dflamand@gmail.com>
Date: Tue, 26 Jul 2016 20:31:14 -0400
Subject: [PATCH 29/37] fixed rand() returning only zero

---
 newlib/libc/stdlib/rand.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/newlib/libc/stdlib/rand.c b/newlib/libc/stdlib/rand.c
index 131e5cf..94f7928 100644
--- a/newlib/libc/stdlib/rand.c
+++ b/newlib/libc/stdlib/rand.c
@@ -84,8 +84,13 @@ _DEFUN_VOID (rand)
      Edition, Addison-Wesley, 1998, p. 106 (line 26) & p. 108 */
   _REENT_CHECK_RAND48(_REENT);
   _REENT_RAND_NEXT(_REENT) = 
+#if RAND_MAX == 32767
+     _REENT_RAND_NEXT(_REENT) * __extension__ 1284865837L + 1;
+  return (int)((_REENT_RAND_NEXT(_REENT) >> 16) & RAND_MAX);
+#else
      _REENT_RAND_NEXT(_REENT) * __extension__ 6364136223846793005LL + 1;
   return (int)((_REENT_RAND_NEXT(_REENT) >> 32) & RAND_MAX);
+#endif
 }
 
 #endif /* _REENT_ONLY */
-- 
2.19.1.windows.1

