From 33f429e861eb5b033b4175b5e4e468fa25a7a6f3 Mon Sep 17 00:00:00 2001
From: Brian Dominy <brian@oddchange.com>
Date: Wed, 24 Jun 2009 19:10:33 -0400
Subject: [PATCH 10/37] Add m6809 to libgloss subdirs.

---
 libgloss/configure.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libgloss/configure.in b/libgloss/configure.in
index 2fa792f..9db5128 100644
--- a/libgloss/configure.in
+++ b/libgloss/configure.in
@@ -42,7 +42,11 @@ case "${target}" in
 	AC_CONFIG_SUBDIRS(m68hc11)
 	config_testsuite=true
 	;;
-  fido-*-* | m68*-*-*)
+  m6809-*-*)
+	AC_CONFIG_SUBDIRS(m6809)
+	config_testsuite=false
+	;;
+  fido-*-* | m68k*-*-*)
 	AC_CONFIG_SUBDIRS(m68k)
 	config_testsuite=true
 	;;
-- 
2.19.1.windows.1

