From 96c48ec58af328cdf0ffe7f4ea0d1f1ac26dc903 Mon Sep 17 00:00:00 2001
From: dfffffff <dflamand@gmail.com>
Date: Mon, 12 Jan 2015 08:18:18 -0500
Subject: [PATCH 14/37] make _POINTER_INT proper size

---
 newlib/libc/include/sys/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/newlib/libc/include/sys/config.h b/newlib/libc/include/sys/config.h
index 0ac2b4e..203b583 100644
--- a/newlib/libc/include/sys/config.h
+++ b/newlib/libc/include/sys/config.h
@@ -42,7 +42,7 @@
 #define _POINTER_INT short
 #endif
 
-#if defined(__mc68hc11__) || defined(__mc68hc12__) || defined(__mc68hc1x__)
+#if defined(__m6809__) || defined(__mc68hc11__) || defined(__mc68hc12__) || defined(__mc68hc1x__)
 #undef INT_MAX
 #undef UINT_MAX
 #define INT_MAX __INT_MAX__
-- 
2.19.1.windows.1

