From 7bd18c52d3068b65b77da36c944a3d6ce9cc43d5 Mon Sep 17 00:00:00 2001
From: Brian Dominy <brian@oddchange.com>
Date: Wed, 24 Jun 2009 16:09:24 -0400
Subject: [PATCH 08/37] Configure jmpbuf size for 6809.

---
 newlib/libc/include/machine/setjmp.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/newlib/libc/include/machine/setjmp.h b/newlib/libc/include/machine/setjmp.h
index 2b7aeb2..8ec3407 100644
--- a/newlib/libc/include/machine/setjmp.h
+++ b/newlib/libc/include/machine/setjmp.h
@@ -44,6 +44,11 @@ _BEGIN_STD_C
 #define _JBLEN  32
 #endif
 
+#if defined(__m6809__)
+/* D, X, Y, U, S, PC, DP, CC, plus some padding */
+#define _JBLEN 10
+#endif
+
 #if defined(__Z8001__) || defined(__Z8002__)
 /* 16 regs + pc */
 #define _JBLEN 20
-- 
2.19.1.windows.1

