From d652360353fa00d9f36309a2ee255c70df438774 Mon Sep 17 00:00:00 2001
From: dfffffff <dflamand@gmail.com>
Date: Sun, 29 May 2016 07:57:37 -0400
Subject: [PATCH 16/37] updated sys/config.h

---
 newlib/libc/include/sys/config.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/newlib/libc/include/sys/config.h b/newlib/libc/include/sys/config.h
index 203b583..e7abdd6 100644
--- a/newlib/libc/include/sys/config.h
+++ b/newlib/libc/include/sys/config.h
@@ -49,6 +49,11 @@
 #define UINT_MAX (__INT_MAX__ * 2U + 1)
 #define _POINTER_INT short
 #endif
+#if defined(__m6809__)
+/*#define _REENT_SMALL  <- code is bigger */
+#define __SINGLE_THREAD__
+#define __SMALL_BITFIELDS
+#endif
 
 #ifdef ___AM29K__
 #define _FLOAT_RET double
-- 
2.19.1.windows.1

